qa/tasks/cephfs/test_subvolume.py

```diff
@@ -0,0 +1,207 @@
+import logging
+
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+from teuthology.exceptions import CommandFailedError
+
+log = logging.getLogger(__name__)
+
+
+class TestSubvolume(CephFSTestCase):
+    CLIENTS_REQUIRED = 1
+    MDSS_REQUIRED = 1
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def setup_test(self):
+        self.mount_a.run_shell(['mkdir', 'group'])
+        self.mount_a.run_shell(['mkdir', 'group/subvol1'])
+        self.mount_a.run_shell(['setfattr', '-n', 'ceph.dir.subvolume',
+                                '-v', '1', 'group/subvol1'])
+        self.mount_a.run_shell(['mv', 'group/subvol1', 'group/subvol2'])
+
+    def cleanup_test(self):
+        self.mount_a.run_shell(['rm', '-rf', 'group'])
+
+    def test_subvolume_move_out_file(self):
+        """
+        To verify that file can't be moved out of subvolume
+        """
+        self.setup_test()
+
+        self.mount_a.run_shell(['touch', 'group/subvol2/file1'])
+
+        # file can't be moved out of a subvolume
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.run_shell(['rename', 'group/subvol2/file1',
+                                    'group/file1', 'group/subvol2/file1'])
+
+        # clean up
+        self.cleanup_test()
```

Comment:
call this in tearDown() as mentioned above.

Likewise for other test methods.

Reply:
now the tests have only one call to self.setup_test() and one call to self.cleanup_test()